integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/resources/kafka/KafkaResource.java

```diff
@@ -0,0 +1,155 @@
+import com.openshift.cloud.api.kas.auth.models.NewTopicInput;
+import com.openshift.cloud.api.kas.auth.models.TopicSettings;
+
+import io.restassured.http.ContentType;
+
+import static io.restassured.RestAssured.given;
+
+public class KafkaResource {
+    public static void createKafkaTopic(String topicName) {
+        KafkaConnectionParameters kafkaConnectionParameters = KafkaConnectionParameters.getInstance();
+        ApiClient apiClient = createApiClient();
+        TopicsApi topicsApi = new TopicsApi(apiClient);
+        NewTopicInput newTopicInput = new NewTopicInput()
+                .name(topicName)
+                .settings(new TopicSettings().numPartitions(1));
+        try {
+            topicsApi.createTopic(newTopicInput);
+        } catch (ApiException e) {
+            throw new RuntimeException("Error creating new topic", e);
+        }
+
+        AclsApi aclsApi = new AclsApi(apiClient);
+        AclBinding aclBindingRead = new AclBinding()
+                .resourceType(AclResourceType.TOPIC)
+                .resourceName(topicName)
+                .patternType(AclPatternType.LITERAL)
+                .principal("User:" + kafkaConnectionParameters.getOpsClientID())
+                .operation(AclOperation.READ)
+                .permission(AclPermissionType.ALLOW);
+        AclBinding aclBindingWrite = new AclBinding()
+                .resourceType(AclResourceType.TOPIC)
+                .resourceName(topicName)
+                .patternType(AclPatternType.LITERAL)
+                .principal("User:" + kafkaConnectionParameters.getOpsClientID())
+                .operation(AclOperation.WRITE)
+                .permission(AclPermissionType.ALLOW);
+        try {
+            aclsApi.createAcl(aclBindingRead);
+            aclsApi.createAcl(aclBindingWrite);
+        } catch (ApiException e) {
+            throw new RuntimeException("Error assigning ACL to topic", e);
+        }
+    }
+
+    public static void deleteKafkaTopic(String topicName) {
+        KafkaConnectionParameters kafkaConnectionParameters = KafkaConnectionParameters.getInstance();
+        ApiClient apiClient = createApiClient();
+        AclsApi aclsApi = new AclsApi(apiClient);
+        try {
+            aclsApi.deleteAcls(AclResourceTypeFilter.TOPIC,
+                    topicName,
+                    AclPatternTypeFilter.LITERAL,
+                    "User:" + kafkaConnectionParameters.getOpsClientID(),
+                    AclOperationFilter.READ,
+                    AclPermissionTypeFilter.ALLOW);
+            aclsApi.deleteAcls(AclResourceTypeFilter.TOPIC,
+                    topicName,
+                    AclPatternTypeFilter.LITERAL,
+                    "User:" + kafkaConnectionParameters.getOpsClientID(),
+                    AclOperationFilter.WRITE,
+                    AclPermissionTypeFilter.ALLOW);
+        } catch (Exception e) {
+            throw new RuntimeException("Error removing ACL for topic", e);
```

Comment:
Same here

Reply:
I'm not sure here. It is more about deleting of ACLS which were already created. Basically all items need to be destroyed.